subprojects/core/src/main/java/org/gradle/api/DefaultTask.java

```diff
@@ -60,7 +60,7 @@ public Project getProject() {
     }
 
     @Override
-    public String getName() {
+    public final String getName() {
```

Comment:
It seems somewhat unmotivated to sprinkle `final` on the methods in `DefaultTask`. I wonder if there is a better way to ensure nobody overrides these methods.

Reply:
I will check if we can do this when we decorate a class or somewhere there